### PR TITLE
Add namespace label selector to namespace dashboard links

### PR DESCRIPTION
**Pull request by spadgett**
https://jira.coreos.com/browse/CONSOLE-1828

/assign @rhamilto 
/cc @jhadvig

### CODE REVIEW

File: frontend/public/components/dashboard/project-dashboard/project-dashboard.tsx [hunk 1]
```diff
@@ -26,13 +26,29 @@ const getNamespaceDashboardConsoleLinks = (
...
+
+      const namespaces: string[] = _.get(link, 'spec.namespaceDashboard.namespaces');
+      const selector: Selector = _.get(link, 'spec.namespaceDashboard.namespaceSelector');
+
+      // If neither namespaces or selector was provided, show the link for all namespaces.
+      if (_.isEmpty(namespaces) && _.isEmpty(selector)) {
+        return true;
+      }
+
+      // Show the link if either namespaces or the selector matches this namespace.
```
Comment (rhamilto) [inline frontend/public/components/dashboard/project-dashboard/project-dashboard.tsx]: Comment is incorret?

Comment (spadgett) [reply frontend/public/components/dashboard/project-dashboard/project-dashboard.tsx]: I think it's right? We show the link if we match either namespaces OR selector. (Maybe I should rephrase.)

Comment (jhadvig) [reply frontend/public/components/dashboard/project-dashboard/project-dashboard.tsx]: The comment is correct but does not reflect the code change, but overall logic.
Check https://github.com/openshift/api/blob/master/console/v1/types_console_link.go#L48

Comment (spadgett) [reply frontend/public/components/dashboard/project-dashboard/project-dashboard.tsx]: Isn't the API doc saying the same thing?

**Pull request merged**